### ISSUE TITLE
balena push fails with fatal error

### ISSUE THREAD

**Issue by LucianBuzzo** [balena-io/balena-cli#2200]
Running balena push and I get this error:
```
[Build]   [ui]  ---> 33f93e83844a
[Build]   [ui] Step 25/25 : LABEL io.resin.local.service=ui
[Build]   [ui]  ---> Running in 2241fe6b3d13
[Build]   [ui] Removing intermediate container 2241fe6b3d13
[Build]   [ui]  ---> 4f7fd7fed3ff
[Build]   [ui] Successfully built 4f7fd7fed3ff
[Build]   [ui] Successfully tagged local_image_ui:latest

[Live]    Waiting for device state to settle...


#
# Fatal error in , line 0
# Check failed: result.second.
#
#
#
#FailureMessage Object: 0x7fffd7d27f30
 1: 0xa71261  [node]
 2: 0x19d3254 V8_Fatal(char const*, ...) [node]
 3: 0xe5b549 v8::internal::GlobalBackingStoreRegistry::Register(std::shared_ptr<v8::internal::BackingStore>) [node]
 4: 0xba6a98 v8::ArrayBuffer::GetBackingStore() [node]
 5: 0x9c1be0 napi_get_typedarray_info [node]
 6: 0x7f9cf4575f60  [/usr/local/lib/node_modules/balena-cli/node_modules/net-keepalive/node_modules/ffi-napi/build/Release/ffi_bindings.node]
 7: 0x7f9cf4579e6b FFI::FFI::InitializeBindings(Napi::Env, Napi::Object) [/usr/local/lib/node_modules/balena-cli/node_modules/net-keepalive/node_modules/ffi-napi/build/Release/ffi_bindings.node]
 8: 0x7f9cf457b0a0  [/usr/local/lib/node_modules/balena-cli/node_modules/net-keepalive/node_modules/ffi-napi/build/Release/ffi_bindings.node]
 9: 0x7f9cf4576034 __napi_Init(napi_env__*, napi_value__*) [/usr/local/lib/node_modules/balena-cli/node_modules/net-keepalive/node_modules/ffi-napi/build/Release/ffi_bindings.node]
10: 0x9d54a8 napi_module_register_by_symbol(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, napi_value__* (*)(napi_env__*, napi_value__*)) [node]
11: 0x9da07f  [node]
12: 0x9d93ab node::binding::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&) [node]
13: 0xbe571b  [node]
14: 0xbe6cc6  [node]
15: 0xbe7346 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [node]
16: 0x1401319  [node]
Illegal instruction (core dumped)
Makefile:434: recipe for target 'push' failed
make: *** [push] Error 132

```

The make command is just a wrapper around `balena push` with additional `--env` params.
This is on WSL in windows 10 using node v14.15.5 and cli v11.23.0

**Comment by pdcastro** [comment]
Thanks for reporting it. CLI v11.23.0 is more than a year old, please update! 😁 
Updating the CLI won't fix it though, as this issue is related to Node.js v14, which is not yet fully supported by the CLI - we are working on it. Since CLI release 12.38.4, running the CLI with Node.js v14 will print this warning message:

```
$ balena push 192.168.1.46
------------------------------------------------------------------------------
Warning: Node version "v14.16.0" does not match required versions ">=10.20.0 <13.0.0".
This may cause unexpected behavior. To upgrade Node, visit:
https://nodejs.org/en/download/
------------------------------------------------------------------------------
[Info]    Starting build on device 192.168.1.46
...
```

In addition to the runtime warning regarding Node.js v14, the version requirement is documented in the CLI's [npm installation instructions](https://github.com/balena-io/balena-cli/blob/master/INSTALL-ADVANCED.md#npm-installation).

Connects-to: #2079, #2165, #1990

cc: @mbalamat

**Comment by pdcastro** [comment]
The `Illegal instruction (core dumped)` issue is probably resolved in CLI v12.44.0 (PR #2168), though I am not sure we actually got to reproduce the error in order to confirm that it was fixed (at least I didn't, maybe @mbalamat did - his commit claims to fix this issue). Note that the CLI, especially `balena push`, still does not support Node.js v14, and in particular issue #2165 remains open.
